conjure/servicewriter.go

```diff
@@ -1018,25 +1018,13 @@ func returnTypesForEndpoint(endpointDefinition spec.EndpointDefinition, info typ
 	var returnTypes []expression.Type
 	imports := make(StringSet)
 	if endpointDefinition.Returns != nil {
-		var goType string
-		returnBinary, err := isReturnTypeSpecificType(endpointDefinition.Returns, visitors.IsBinary)
+		typer, err := visitors.NewConjureTypeProviderTyper(*endpointDefinition.Returns, info)
 		if err != nil {
 			return nil, nil, err
 		}
-		if returnBinary {
-			// special case: "binary" type resolves to []byte in structs, but indicates a streaming response when
-			// specified as the return type of a service, so use "io.ReadCloser".
-			goType = types.IOReadCloserType.GoType(info)
-			imports.AddAll(NewStringSet(types.IOReadCloserType.ImportPaths()...))
```

Comment:
The refactor here doesn't really look like 1:1 -- for example, don't see anything in the newly refactored code that adds the import if needed.

Can you add the functionality in a way that's more obvious without the refactor? I would have expected to see logic that as essentially along the lines of "if the return type if an optional of binary, then special case and return the type that is pointer to IO.ReadCloser"

Reply:
Above comment still stands -- don't understand why this PR would modify the "else" clause?

Reply:
I'd prefer to minimize the amount/scope of the special casing here. Part of what allowed this bug to creep in was that the `binary` case was treated handled separately when in reality the binary case is identical to all other cases except we replace references to `[]byte` with `ReadCloser` inside of the `typer` variable.